noir_stdlib/src/hash/keccak.nr

```diff
@@ -39,13 +39,11 @@ pub(crate) fn keccak256<let N: u32>(input: [u8; N], message_size: u32) -> [u8; 3
     // means we need to swap our byte ordering
     let num_limbs = max_blocks * LIMBS_PER_BLOCK; //max_blocks_length / WORD_SIZE;
     for i in 0..num_limbs {
-        let mut temp = [0; WORD_SIZE];
         let word_size_times_i = WORD_SIZE * i;
-        for j in 0..WORD_SIZE {
-            temp[j] = block_bytes[word_size_times_i+j];
-        }
-        for j in 0..WORD_SIZE {
-            block_bytes[word_size_times_i + j] = temp[7 - j];
+        for j in 0..WORD_SIZE / 2 {
+            let temp = block_bytes[word_size_times_i + 7 - j];
+            block_bytes[word_size_times_i + 7 - j] = block_bytes[word_size_times_i+j];
```

Comment:
Maybe:

```suggestion
            let temp = block_bytes[word_size_times_i + WORD_SIZE - 1 - j];
            block_bytes[word_size_times_i + WORD_SIZE - 1 - j] = block_bytes[word_size_times_i+j];
```

Reply:
(it might make things clearer, hopefully not slower, because until I checked that WORD_SIZE was 8 I didn't know why the refactor would work with 7)

Reply:
We could also add another variable like `word_size_times_i` that is equal to `word_size_times_i + (WORD_SIZE - 1)`.

Reply:
I've renamed these to `limb_start` and `limb_end` to be more clearer about why we care about these.